internal/charmstore/search_test.go

```diff
@@ -804,3 +837,75 @@ func (s *StoreSearchSuite) TestUpdateConflict(c *gc.C) {
 	c.Assert(err, gc.Equals, nil)
 	c.Assert(updated, gc.Equals, false)
 }
+
+func (s *StoreSearchSuite) TestMultiSeriesCharmFiltersSeriesCorrectly(c *gc.C) {
+	charmArchive := storetesting.Charms.CharmDir("juju-gui")
+	url := newResolvedURL("cs:~charmers/juju-gui-25", -1)
+	err := s.store.AddCharmWithArchive(url, charmArchive)
+	c.Assert(err, gc.IsNil)
+	bURL := baseURL(&url.URL)
+	baseEntity, err := s.store.FindBaseEntity(bURL)
+	c.Assert(err, gc.IsNil)
+	baseEntity.ACLs.Read = append(baseEntity.ACLs.Read, params.Everyone)
+	err = s.store.DB.BaseEntities().UpdateId(baseEntity.URL, baseEntity)
+	c.Assert(err, gc.IsNil)
+	err = s.store.UpdateSearch(url)
+	c.Assert(err, gc.IsNil)
+	s.store.ES.Database.RefreshIndex(s.TestIndex)
+	res, err := s.store.Search(SearchParams{
+		Filters: map[string][]string{
+			"name": []string{"juju-gui"},
+		},
+	})
+	c.Assert(err, gc.IsNil)
+	c.Assert(res.Results, gc.HasLen, 1)
+	c.Assert(res.Results[0].URL.String(), gc.Equals, url.String())
+	res, err = s.store.Search(SearchParams{
+		Filters: map[string][]string{
+			"name":   []string{"juju-gui"},
+			"series": []string{"trusty"},
+		},
+	})
+	c.Assert(err, gc.IsNil)
+	c.Assert(res.Results, gc.HasLen, 1)
+	c.Assert(res.Results[0].URL.String(), gc.Equals, url.String())
+	res, err = s.store.Search(SearchParams{
+		Filters: map[string][]string{
+			"name":   []string{"juju-gui"},
+			"series": []string{"vivid"},
+		},
+	})
+	c.Assert(err, gc.IsNil)
+	c.Assert(res.Results, gc.HasLen, 1)
+	c.Assert(res.Results[0].URL.String(), gc.Equals, url.String())
+	res, err = s.store.Search(SearchParams{
+		Filters: map[string][]string{
+			"name":   []string{"juju-gui"},
+			"series": []string{"saucy"},
+		},
+	})
+	c.Assert(err, gc.IsNil)
+	c.Assert(res.Results, gc.HasLen, 0)
+}
+
+func (s *StoreSearchSuite) TestMultiSeriesCharmSortsSeriesCorrectly(c *gc.C) {
+	charmArchive := storetesting.Charms.CharmDir("juju-gui")
+	url := newResolvedURL("cs:~charmers/juju-gui-25", -1)
+	err := s.store.AddCharmWithArchive(url, charmArchive)
+	c.Assert(err, gc.IsNil)
+	bURL := baseURL(&url.URL)
+	baseEntity, err := s.store.FindBaseEntity(bURL)
+	c.Assert(err, gc.IsNil)
+	baseEntity.ACLs.Read = append(baseEntity.ACLs.Read, params.Everyone)
+	err = s.store.DB.BaseEntities().UpdateId(baseEntity.URL, baseEntity)
```

Comment:
Why do we need to do this? And given that we're doing it, why not do it at the same level as other tests and use s.store.SetPerms ?